File: PHP/FindTheIndexOfTheFirstOccurrenceInStringTest.php
<?php
/**
 * @version PHP 8.2.0
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ ./vendor/bin/phpunit LengthOfLastWordTest.php
 */
namespace PHP;

use PHPUnit\Framework\TestCase;
require ("FindTheIndexOfTheFirstOccurrenceInString.php");

class FindTheIndexOfTheFirstOccurrenceInStringTest extends TestCase {

    public function testFindTheIndexOfTheFirstOccurrenceInString() {
        $obj = new FindTheIndexOfTheFirstOccurrenceInString();
        //Test 1
        $want = 0;
        $got = $obj->strStr( "sadbutsad", "sad" );
        echo "\nTest 1: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
        //Test 2
        $want = -1;
        $got = $obj->strStr( "leetcode", "leeto" );
        echo "\nTest 2: retornou " . $got . " == esperado: " . $want;
        $this->assertEquals($want, $got);
    }
}

File: GO/FindTheIndexOfTheFirstOccurrenceInString_test.go
/**
 * @version GO 1.20.2
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ go test -timeout 999999s -run TestFirstMissingPositive
 */
package GO

import (
	"reflect"
	"testing"
)

func TestFindTheIndexOfTheFirstOccurrenceInStringTest(t *testing.T) {
	tables := []struct {
		want int
		word string
		look string
	}{
		{0, "sadbutsad", "sad"},
		{-1, "leetcode", "leeto"},
	}

	for _, table := range tables {
		got := strStr(table.word, table.look)

		if !reflect.DeepEqual(got, table.want) {
			t.Errorf("Waiting for this %d but the return was this: %d", table.want, got)
		}
	}
}
